axlearn/experiments/text/gpt/fuji.py

```diff
@@ -293,6 +296,45 @@ def get_trainer_kwargs(
                 ),
             ),
         )
+    elif model_size == "405B":
+        trainer_kwargs = dict(
+            model_kwargs=dict(
+                num_layers=126,
+                hidden_dim=53248,
+                num_heads=128,
+                # No GQA support in V1 models, so num_kv_heads is the same as num_heads.
+                num_kv_heads=None if version == Version.V1 else 8,
+                rope_theta=rope_theta,
+                flash_attention=flash_attention,
+            ),
+            learner_kwargs=dict(peak_lr=8e-5, weight_decay=0.1),
+            max_sequence_length=max_sequence_length,
+            train_batch_size=train_batch_size,
```

Comment:
Training batch is not 4M tokens for 400B model, it is 16M according to [this](https://scontent-sjc3-1.xx.fbcdn.net/v/t39.2365-6/463020162_522238820565582_8192401983671993921_n.pdf?_nc_cat=108&ccb=1-7&_nc_sid=3c67a6&_nc_ohc=6-qU4tNAGO0Q7kNvgFEd45q&_nc_zt=14&_nc_ht=scontent-sjc3-1.xx&_nc_gid=AEgnKoGUXaRC0u-92CIp6yA&oh=00_AYANniN5JSP4kUFXlEHhCHLd-nBRxuZkqQVAd1x97HCkiw&oe=671F1D59)

Reply:
fixed